pkg/virt-handler/hotplug-disk/mount.go

```diff
@@ -663,6 +667,10 @@ func (m *volumeMounter) Unmount(vmi *v1.VirtualMachineInstance) error {
 			var err error
 			if m.isBlockVolume(&vmi.Status, volumeStatus.Name) {
 				path, err = safepath.JoinNoFollow(basePath, volumeStatus.Name)
+				if errors.Is(err, os.ErrNotExist) {
+					// already unmounted or never mounted
+					continue
+				}
 			} else if m.isDirectoryMounted(&vmi.Status, volumeStatus.Name) {
 				path, err = m.hotplugDiskManager.GetFileSystemDirectoryTargetPathFromHostView(virtlauncherUID, volumeStatus.Name, false)
 				if os.IsExist(err) {
```

Comment:
Not related to the PR, but at first look this does not seem right. Shouldn't this rather check if the directory does not exist and then continue?
```suggestion
				if errors.Is(err, os.ErrNotExist) {
```
Here `GetFileSystemDirectoryTargetPathFromHostView` has `create` argument set to `false` hence it should not return "file already exists" error.